### DISCUSSION TITLE
Shortcuts for halfs in each display

### DISCUSSION THREAD

**Discussion by lprsd** [rxhanson/Rectangle#476] [Show and tell]
- Great app with a lot of pre-built thoughtful shortcuts. Thanks!
- To reduce my cognitive load, to where something should be, my mental model is one of the six spaces in the 3 screens.
- I have 3 screens like in the image below and exactly 2 windows split in half in each.
- I want to set up a keyboard shortcut to take the given window to one of those 6 spaces.

![IMG_20210525_141236](https://user-images.githubusercontent.com/97380/119472368-ce93b580-bd67-11eb-9075-9a8f9ac3e4a7.jpg)

Right now I guess, I will need to use 2 keystrokes to achieve it. But if I only need these choices and none else, anyway it can be achieved?

Thanks for hearing me!

**Comment by rxhanson** [comment]
I appreciate that you took the time to propose the idea. It's not something I have plans to add to Rectangle, but it is something would be covered by a feature I have in my backlog for [Hookshot](https://hookshot.app) (my paid superset of Rectangle). The feature I am planning on adding is the ability to select a display in the custom shortcut builder. I'll make a note to report back here when that feature is in Hookshot, if you're interested (no guarantees on a timeframe, though).

**Reply by rxhanson** [reply]
Just a quick heads up that I have added specifying a display in the custom shortcut builder in the latest version of [Hookshot](https://hookshot.app), so I think that this covers your use case? 

Either way, feel free to let me know if you are looking to implement something in a fork of Rectangle and have any questions.

**Reply by lprsd** [reply]
Thank you. Looks great.

I just bought a license and configured it with these shortcuts.

<img width="388" alt="Screenshot 2021-06-08 at 11 25 23 AM" src="https://user-images.githubusercontent.com/97380/121130738-728f5d80-c84c-11eb-8c55-4e3bebb8eb34.png">

I'm also going to be sharing about it more including here: https://apple.stackexchange.com/questions/421214/window-layout-management-keyboard-shortcut

Given that I can achieve what I want now, maybe I wouldn't need to dive into the Rectangle code base after all. Thanks.

**Reply by rxhanson** [reply]
Awesome, thanks for sharing it on the apple stackexchange!

**Comment by lprsd** [comment]
Thank you.

I will look forward to the update.

**Comment by lprsd** [comment]
Thanks Ryan.

- I was checking out the code of Rectangle today.
- Seems to me like "Half of (one of) the display" can be a core use case and can be implemented rather easily too. (Not sure if thirds, fourths, the sixths are really used that much.)
- Settings can provide an option for toggling this feature.
- Ctrl+Opt+[1-9,a-f] can provide automatic (or default option with NSShortcut) binding.

If I take a shot at implementing this, would you consider merging it as a PR?

If you're not keen, I plan to implement just only this feature as an independent app. (By referring to the relevant material from this, the spectacle project, and the documentation of course.)

Thanks again for your time reviewing my thoughts.

**Comment by rxhanson** [comment]
> Seems to me like "Half of (one of) the display" can be a core use case and can be implemented rather easily too. (Not sure if thirds, fourths, the sixths are really used that much.)

The shortcuts that are in Rectangle are all there either to match what was in Spectacle, or by popular demand. Thirds are apparently a very popular use case, and fourths and sixths were added due to high demand with a lot more people now using ultra wide displays. Honestly, I have never really liked adding more shortcuts because Rectangle is intended to be a simple app, and I only add something in if it's really wanted. I've also not been excited about adding in an "advanced settings" toggle for the same reason. Keeping it simple makes it a great starter project for anyone wanting to take it and make their own window manager. With that said, I do welcome ideas and don't like shooting down contributions.  Also, there are a lot of people that want to execute a shortcut to move a window to  a specified display - which sounds like it is half of your need. 

> Settings can provide an option for toggling this feature.
Ctrl+Opt+[1-9,a-f] can provide automatic (or default option with NSShortcut) binding.

As mentioned, I have purposefully left out "advanced settings", but I'd be willing to roll it in if it's done well. My thought would be a "hold the option key and the 'Preferences...' menu item turns into 'Advanced Preferences...' and that opens a different window with the different shortcuts. Of course, anything added in here would not be enabled by default without the user setting it.

> If I take a shot at implementing this, would you consider merging it as a PR?

I would consider a PR, but I'm actually not 100% sure I know what the end result would be that you're looking to add. My recommendation is to just fork Rectangle and build what you want; then we can see if it makes sense to merge in. I'd love to see what you build, even if it's not something you end up wanting to merge into Rectangle.
